pkg/types/message_envelope.go

```diff
@@ -164,3 +188,53 @@ func fromContext(ctx context.Context, key string) string {
 	}
 	return hdr
 }
+
+func ConvertPayloadToByteArray(payload any) ([]byte, error) {
+	switch v := payload.(type) {
+	case []byte:
+		return v, nil
+	case string:
+		return []byte(v), nil
+	default:
+		return json.Marshal(v)
+	}
+}
+
+func GetPayload[T any](payload any) (res T, err error) {
```

Comment:
```suggestion
func GetMsgPayload[T any](msg MessageEvelope) (res T, err error) {
```
check the content type. if it's CBOR, use cbor.Marshal instead of json.Marshal